pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -240,6 +240,7 @@ public void doRemoveExpiredPrimaryKeys() {
     persistWatermark(_largestSeenComparisonValue);
   }
 
+  // doAddRecord returns whether the event is out-of-order or not
```

Comment:
Don't add javadoc here. Let's add it into the base class (and also the interface), and follows the javadoc format.